### ISSUE TITLE
Maintaining GitHub Enterprise relative to GitHub.com

### ISSUE THREAD

**Issue by djch** [mdo/ama#157]
You might not be the best GitHubber to ask about this, but could you shed any light on how the repo for GitHub Enterprise is maintained relative to GitHub.com?
- Is GHE a fork of github.com that gets the same commits merged into it?
- If so, at what point does code get shipped from one to the other?
- Anything else interesting done to maintain the differences between each code base? (Looks like GHE might have some unique CSS going on, for example)

**Comment by mdo** [comment]
Hey there! I don't work on it directly anymore, but I started out at GitHub working on the Enterprise team. Mostly on the marketing site, Enterprise tools, etc. These days we've got a few more folks working on it and I spend my days on GitHub.com.

> Is GHE a fork of github.com that gets the same commits merged into it?

Not exactly. Enterprise is a separate mode that we can run from our dev environments and is distributed as a VM of sorts. It's the same code base, but with additional tools around it.

> If so, at what point does code get shipped from one to the other?

They do releases every few weeks I think and cut a branch for it. The changes that make their way into that branch are frozen awhile before the release is shipped.

> Anything else interesting done to maintain the differences between each code base? (Looks like GHE might have some unique CSS going on, for example)

Almost all of the GHE related frontend stuff is already in the dotcom codebase. It just doesn't get served until you move into Enterprise mode.
